supabase/migrations/20250910120000_create_files_table.sql

```diff
@@ -0,0 +1,63 @@
+create table if not exists "public"."files" (
+    "id" uuid not null default gen_random_uuid(),
+    "owner_account_id" uuid not null,
+    "artist_account_id" uuid not null,
+    "storage_key" text not null,
+    "file_name" text not null,
+    "mime_type" text,
+    "size_bytes" bigint,
+    "description" text,
+    "tags" text[] default '{}'::text[],
+    "shared_with_artist_account_ids" uuid[] default '{}'::uuid[],
+    "shared_with_account_ids" uuid[] default '{}'::uuid[],
+    "shared_with_all_artist_members" boolean not null default false,
```

Comment:
How will these `shared` fields be used?
- arrays are hard to query on.
- typically, we'd either create multiple entries or create a new table which is easier to query this.

Reply:
For example, how would I query all files shared with my account / artist?

Reply:
Additionally, none of these `shared` fields are foreign keys to the `accounts` table.